src/grid_overlay.cpp

```diff
@@ -216,6 +225,27 @@ void resetCounts (OverlayClouds* overlay)
   fill(overlay->pass_through_counts.begin(), overlay->pass_through_counts.end(), 0);
 }
 
+void saturateCounts (OverlayClouds& overlay)
+{
+  saturateCounts(overlay, 0);
+}
+
+void saturateCounts (OverlayClouds& overlay, const size_t steps)
+{
+  for (size_t i = 0; i < overlay.hit_counts.size(); ++i)
+  {
+    const unsigned hit_count = overlay.hit_counts[i];
+    const unsigned pass_through_count = overlay.pass_through_counts[i];
+
+    const double occupancy_threshold = overlay.occupancy_threshold;
+    const double hit_threshold = pass_through_count * occupancy_threshold;
+
+    overlay.hit_counts[i] = hit_count > hit_threshold ?
+                            hit_threshold + steps + 1 :
```

Comment:
Note that `hit_threshold + steps` is actually a simplification of `(pass_through_count + steps / occupancy_threshold) * occupancy_threshold`, where `steps / occupancy_threshold` is indeed `steps * increment(occupancy_threshold)`